pravega-video/src/index.rs

```diff
@@ -269,42 +278,49 @@ impl<R: Read + Seek + CurrentHead> IndexSearcher<R> {
             }
 
             // Get last record.
-            if timestamp > PravegaTimestamp::MIN {
-                let index_offset = self.reader.seek(SeekFrom::Start(tail_offset - IndexRecord::RECORD_SIZE as u64))?;
-                // TODO: Below may fail due to https://github.com/pravega/pravega-client-rust/issues/163.
-                let index_record = index_record_reader.read(&mut self.reader)?;
-                // Return last record if desired timestamp is after or equal to it.
-                if index_record.timestamp <= timestamp {
-                    return Ok((index_record, index_offset));
-                }
+            let mut last_index_offset = self.reader.seek(SeekFrom::Start(tail_offset - IndexRecord::RECORD_SIZE as u64))?;
+            // TODO: Below may fail due to https://github.com/pravega/pravega-client-rust/issues/163.
+            let mut last_index_record = index_record_reader.read(&mut self.reader)?;
+            // Return last record if desired timestamp is after or equal to it.
+            if last_index_record.timestamp <= timestamp {
+                return Ok((last_index_record, last_index_offset));
             }
 
             // Read first record.
-            let mut prev_index_offset = self.reader.seek(SeekFrom::Start(first_index_offset))?;
-            let mut prev_index_record = index_record_reader.read(&mut self.reader)?;
+            let mut first_index_offset = self.reader.seek(SeekFrom::Start(first_index_offset))?;
+            let mut first_index_record = index_record_reader.read(&mut self.reader)?;
             // Return first record if desired timestamp is before or equal to it.
-            if timestamp <= prev_index_record.timestamp {
-                return Ok((prev_index_record, prev_index_offset));
+            if timestamp <= first_index_record.timestamp {
+                return Ok((first_index_record, first_index_offset));
             }
 
-            // Read index from the beginning until we find a timestamp greater or equal to the desired timestamp.
-            // TODO: Use binary search or Newton's method.
```

Comment:
Change this comment to "Use binary search algorithm" so that readers understand your intention.

Reply:
Done